### PR TITLE
Set up Nix CI

### PR DESCRIPTION
**Pull request by t4ccer**

### CODE REVIEW

File: purescript.cabal [hunk 1]
```diff
@@ -119,7 +119,7 @@ common defaults
     TypeFamilies
     ViewPatterns
   build-tool-depends:
-    happy:happy ==1.20.0
+    happy:happy ^>= 1.20.0
```
Comment (t4ccer) [inline purescript.cabal:122]: haskell.nix couldn't understand it and it was throwing later at build time on version missmatch

File: nix/plutarch/lib.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{ fetchFromGitHub
+, mkHaskellPackage
+, applyPatches
+, fetchpatch
+}:
+
+let
+  plutarchPackage = applyPatches {
+    name = "plutarch-patched";
```
Comment (klntsky) [inline nix/plutarch/lib.nix:9]: I don't think we need plutarch, since we are targeting PIR, do we?

File: nix/haskell/mk-hackage.nix [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            then head lines
+            else builtins.abort "Could not find line with prefix ''${field}:' in ${cabalPath}";
+        in
+        replaceStrings [ " " ] [ "" ] (head (tail (splitString ":" line)));
+      pname = parse "name";
+      version = parse "version";
+    in
+    { inherit src pname version; };
+
+  mkHackageDir = { pname, version, src }:
```
Comment (klntsky) [inline nix/haskell/mk-hackage.nix:33]: Where is this from? Please put the link so that we know where to look when something breaks

Comment (t4ccer) [reply nix/haskell/mk-hackage.nix:33]: I'm not sure honestly, it's something that we've been copy-pasting around for years at this point. Probably https://github.com/mlabs-haskell/mlabs-tooling.nix/blob/main/mk-hackage.nix is a good reference point but it's not maintained

**Review by klntsky — APPROVED** [review]

**Pull request merged**